File: 5-Make The String Great/solution.cpp
#include <bits/stdc++.h>
using namespace std;

class Solution {
public:
    string makeGood(string s) {
        stack<char> stack;

        for (char c : s) {
            if (stack.empty()) {
                stack.push(c);
            }
            else {
                if ((isupper(c) && stack.top() == tolower(c)) || (islower(c) && stack.top() == toupper(c))) {
                    stack.pop();
                }
                else {
                    stack.push(c);
                }
            }
        }
        
        string result;
        while (!stack.empty()) {
            result = stack.top() + result;
            stack.pop();
        }
        
        return result;
    }
};

File: 5-Make The String Great/solution.py
class Solution:
    def makeGood(self, s: str) -> str:
        stack = []

        for i in s:
            if not stack:
                stack.append(i)
            else:
                if (i.isupper() and stack[-1] == i.lower()) or (i.islower() and stack[-1] == i.upper()):
                    stack.pop()
                else:
                    stack.append(i)

        return "".join(stack)